### DISCUSSION TITLE
Opposite of `JSONL` command

### DISCUSSION THREAD

**Discussion by jqnatividad** [dathere/qsv#261] [Q&A] [answered]
Reposting #209 here for the benefit of folks who are looking for the same thing;

Per @kyku:
> Would it possible to add the inverse to the jsonl subcommand? That is, given a CSV file, convert it to JSONL with JSON keys taken from CSV headers and values taken from data.

**Comment by jqnatividad** [comment]
Per #209:

> @kyku, you can already do that with jq
> 
> https://github.com/stedolan/jq/wiki/Cookbook#convert-a-csv-file-with-headers-to-json
> 
> and you should be able to readily compose pipelines that combines qsv and jq (or other CLI tools for that matter).

**Comment by jqnatividad** [comment, marked as answer]
Starting with version 0.59.0, qsv now has the [`tojsonl`](https://github.com/jqnatividad/qsv/blob/master/src/cmd/tojsonl.rs#L11) command.

**Reply by aborruso** [reply]
Thank you very much

**Comment by kyku** [comment]
@jqnatividad, thanks a lot, having this all in one place can come in handy, indeed.

**Reply by jqnatividad** [reply]
@kyku, checkout v0.60.0 - it has a smarter `tojsonl` - i.e. it infers the JSON data types automagically by running `schema`,`stats` and `frequency` against the CSV first (taking just a few seconds, even for very large files), instead of treating everything as a String JSON type.
